File: python/Stack/stack/filemanager.py
import json
from .Shelf import Shelf 
from .Book import Book
class Filemanager:
    def serialize(self, theshelf ):
        datalist = []
        for i in range (0, theshelf.top+1):
            data = {
                "title": theshelf.books[i].title,
                "author": theshelf.books[i].author
            }
            datalist.append(data)
        with open("data.json", "w") as file:
            json.dump(datalist, file)

    def deserialize(self):
        theself = Shelf()
        with open("data.json", "r") as file:
            datalist = json.load(file)
            for data in datalist:
                book = Book(data["title"], data["author"])
                theself.push(book)
        return theself                
        

File: c++/stack/filemanager.cpp
#include "shelf.cpp"
class FileManager
{
public:
    bool serialize(Shelf theshelf)
    {
        bool status = false;
        ofstream fout("shelf.txt");
        if (fout.is_open())
        {
            for (int i = 0; i <= theshelf.top; i++)
            {
                fout<<theshelf.book[i].id<<"|"<<theshelf.book[i].title<<"|"<< theshelf.book[i].author<<endl ;
                
                status = true;
            }
            fout.close();
            cout << "successfull save " << endl;
        }
        else
        {
            cout << "something wrong" << endl;
        }

        return status;  
    }

    Shelf loadFromFile()
    {
        Book book;
        Shelf theshelf;
        ifstream in("shelf.txt");
        string line;
        while (getline(in, line))
        {
            if (!line.empty())
                theshelf.push(book.deserialize(line));
        }
        return theshelf;
    }
};
